Tests/kaas/kaas-sonobuoy-tests/scs_k8s_conformance_tests/scs_0217_cluster_hardening_test.go

```diff
@@ -0,0 +1,825 @@
+			if err != nil {
+				t.Fatal("failed to list control plane nodes:", err)
+			}
+
+			for _, node := range nodes.Items {
+				nodeIP := node.Status.Addresses[0].Address
+
+				// Check that the API server port (6443) is accessible
+				checkPortAccessibility(t, nodeIP, apiServerPort, true)
+
+				// Check that the control plane ports (other than API server) are not accessible
+				checkPortAccessibility(t, nodeIP, controllerManagerPort, false)
+				checkPortAccessibility(t, nodeIP, schedulerPort, false)
+				checkPortAccessibility(t, nodeIP, kubeletApiPort, false)
+
+				// Check the etcd ports (2379-2380) are not accessible
+				for port := etcdPortRangeStart; port <= etcdPortRangeEnd; port++ {
+					checkPortAccessibility(t, nodeIP, fmt.Sprintf("%d", port), false)
+				}
+			}
+			return ctx
+		})
+
+	testenv.Test(t, f.Feature())
+}
+
+// Test_K8s_Endpoints_HTTPS checks if all Kubernetes endpoints are secured via HTTPS.
+func Test_scs_0217_sonobuoy_K8s_Endpoints_HTTPS(t *testing.T) {
+	f := features.New("Kubernetes endpoint security").Assess(
+		"All Kubernetes endpoints should be secured via HTTPS",
+		func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			// Get the list of endpoints in the cluster
+			endpoints := &corev1.EndpointsList{}
+			err := cfg.Client().Resources().List(context.TODO(), endpoints)
+			if err != nil {
+				t.Fatal("failed to list endpoints:", err)
+			}
+
+			// Iterate over the endpoints and check if they're secured via HTTPS
+			for _, ep := range endpoints.Items {
+				for _, subset := range ep.Subsets {
+					for _, addr := range subset.Addresses {
+						nodeIP := addr.IP
+						for _, port := range subset.Ports {
+							portName := port.Name
+							portNum := port.Port
+
+							// Check if the endpoint is secured via HTTPS
+							if isPortSecuredWithHTTPS(nodeIP, portNum, connectionTimeout) {
+								t.Logf("Endpoint %s:%d (%s) is secured via HTTPS", nodeIP, portNum, portName)
+							} else {
+								t.Errorf("Error: Endpoint %s:%d (%s) is not secured via HTTPS", nodeIP, portNum, portName)
+							}
+						}
+					}
+				}
+			}
+			return ctx
+		})
+
+	testenv.Test(t, f.Feature())
+}
+
+// Test_scs_0217_sonobuoy_Kubelet_HTTPS_Anonymous_Auth_Disabled checks
+// if the Kubelet's anonymous access is disabled based on the configz file.
+func Test_scs_0217_sonobuoy_Kubelet_HTTPS_Anonymous_Auth_Disabled(t *testing.T) {
+	f := features.New("kubelet security").Assess(
+		"Kubelet HTTPS anonymous access should be disabled based on the configz",
+		func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			restConf, err := rest.InClusterConfig()
+			if err != nil {
+				t.Fatal("failed to create rest config:", err)
+			}
+
+			kubeClient, err := kubernetes.NewForConfig(restConf)
+			if err != nil {
+				t.Fatal("failed to create Kubernetes client:", err)
+			}
+
+			nodeList, err := kubeClient.CoreV1().Nodes().List(context.TODO(), v1.ListOptions{})
```

Comment:
there are several places where "context.TODO()" is used. Please change that to "context.Background()".
That should work everywhere and is not as "ugly" as TODO